source/vir/src/interpreter.rs

```diff
@@ -1655,7 +1655,7 @@ fn eval_expr_launch(
         fun_calls: HashMap::new(),
     };
     // Don't run for too long
-    let max_iterations = (rlimit as f64 * RLIMIT_MULTIPLIER as f64) as u64 * RLIMIT_MULTIPLIER;
+    let max_iterations = (rlimit as f64 * RLIMIT_MULTIPLIER as f64) as u64;
```

Comment:
Does this need a check for overflow?

Reply:
I don't think so, the float multiplication will clip at infinity which is turned into u64::MAX by the conversion

Reply:
> I don't think so, the float multiplication will clip at infinity which is turned into u64::MAX by the conversion

Would the same reasoning apply to `(rlimit * RLIMIT_PER_SECOND).min(u32::MAX as f32) as u32`?